### PR TITLE
kanit-font: init at 2020-06-16

### PR DESCRIPTION
**Pull request by toastal**
###### Motivation for this change

There's not enough Thai font options in Nix and [Kanit](https://cadsondemak.github.io/kanit/) is open source and doesn't appear to be in `tlwg`. The build checks out the Kanit git repo from GitHub and copies the prebuilt files into appropriate folders (they don't have "releases"). I'm mostly drawing how this is done by looking at similar fonts. The part that is confusing to me is `name` vs `pname` and if/how `-font` should be added; it doesn't seem too consistent.

```console
$ tree
/nix/store/clqii78x1y81z4fwghksp62ckw4vny3p-kanit-2020-06-16/
└── share
    ├── doc
    │   └── kanit
    │       ├── BRIEF.md
    │       ├── css
    │       │   └── fonts.css
    │       ├── features.html
    │       └── OFL.txt
    └── fonts
        ├── opentype
        │   ├── Kanit-BlackItalic.otf
        │   ├── Kanit-Black.otf
        │   ├── Kanit-BoldItalic.otf
        │   ├── Kanit-Bold.otf
        │   ├── Kanit-ExtraBoldItalic.otf
        │   ├── Kanit-ExtraBold.otf
        │   ├── Kanit-ExtraLightItalic.otf
        │   ├── Kanit-ExtraLight.otf
        │   ├── Kanit-Italic.otf
        │   ├── Kanit-LightItalic.otf
        │   ├── Kanit-Light.otf
        │   ├── Kanit-MediumItalic.otf
        │   ├── Kanit-Medium.otf
        │   ├── Kanit-Regular.otf
        │   ├── Kanit-SemiBoldItalic.otf
        │   ├── Kanit-SemiBold.otf
        │   ├── Kanit-ThinItalic.otf
        │   └── Kanit-Thin.otf
        └── truetype
            ├── Kanit-BlackItalic.ttf
            ├── Kanit-Black.ttf
            ├── Kanit-BoldItalic.ttf
            ├── Kanit-Bold.ttf
            ├── Kanit-ExtraBoldItalic.ttf
            ├── Kanit-ExtraBold.ttf
            ├── Kanit-ExtraLightItalic.ttf
            ├── Kanit-ExtraLight.ttf
            ├── Kanit-Italic.ttf
            ├── Kanit-LightItalic.ttf
            ├── Kanit-Light.ttf
            ├── Kanit-MediumItalic.ttf
            ├── Kanit-Medium.ttf
            ├── Kanit-Regular.ttf
            ├── Kanit-SemiBoldItalic.ttf
            ├── Kanit-SemiBold.ttf
            ├── Kanit-ThinItalic.ttf
            └── Kanit-Thin.ttf
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by toastal** [comment]
Okay, I think I'm done with modifications.

File: pkgs/data/fonts/kanit/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "kanit";
+  version = "2020-06-16";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/kanit/default.nix:5]:
```suggestion
  version = "unstable-2020-06-16";
```

File: pkgs/data/fonts/kanit/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchFromGitHub {
+    owner = "cadsondemak";
+    repo = pname;
+    rev = "467dfe842185681d8042cd608b8291199dd37cda";
+    sha256 = "0p0klb0376r8ki4ap2j99j7jcsq6wgb7m1hf3j1dkncwm7ikmg3h";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
+
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/kanit/default.nix:15]:
```suggestion
```

File: pkgs/data/fonts/kanit/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  installPhase = ''
+    mkdir -p $out/share/doc/${pname}/css/
+    mkdir -p $out/share/fonts/{opentype,truetype}
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/kanit/default.nix:18]:
```suggestion
    mkdir -p $out/share/doc/${pname}/css $out/share/fonts/{opentype,truetype}
```

File: pkgs/data/fonts/kanit/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    cp $src/OFL.txt $src/documentation/{BRIEF.md,features.html} $out/share/doc/${pname}
+    cp $src/documentation/css/fonts.css $out/share/doc/${pname}/css
+    cp $src/fonts/otf/*.otf $out/share/fonts/opentype
+    cp $src/fonts/ttf/*.ttf $out/share/fonts/truetype
+  '';
+
+  meta = with lib; {
+    homepage = "http://cadsondemak.github.io/kanit/";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/kanit/default.nix:27]:
```suggestion
    homepage = "https://cadsondemak.github.io/kanit/";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/kanit/default.nix (hunk 1 above)
Comment (toastal) [reply pkgs/data/fonts/kanit/default.nix:5]: Seems reasonable

**Pull request merged**